src/api/src/services/BinaryFlashingStrategy/AmazonS3/index.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable import/prefer-default-export */
+import fetch, { Headers } from 'node-fetch';
+import { writeFile } from 'fs/promises';
+import fs from 'fs';
+
+export const downloadIfModified = async (
+  url: string,
+  outputFile: string
+): Promise<boolean> => {
+  const headers = new Headers();
+  if (fs.existsSync(outputFile)) {
+    const stat = await fs.promises.stat(outputFile);
+    headers.append('If-Modified-Since', stat.mtime.toUTCString());
+  }
+
+  const response = await fetch(url, { headers });
+  console.log(response);
+
+  if (response.status === 304) {
+    return false;
+  }
+
+  if (response.status === 200) {
+    const buffer = await response.buffer();
+    await writeFile(outputFile, buffer);
+    return true;
+  }
+
+  throw new Error(`${url} is not available`);
```

Comment:
I think it would be better to include actual HTTP status code / error message in the error message. It will help with the debugging in the future. Especially with user issues like unreliable network connection / etc.